streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -509,38 +517,64 @@ void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
             prepareCommitAndAddOffsetsToMap(commitNeededActiveTasks, consumedOffsetsPerTask);
         }
 
-        // even if commit failed, we should still continue and complete suspending those tasks,
-        // so we would capture any exception and throw
+        // even if commit failed, we should still continue and complete suspending those tasks, so we would capture
+        // any exception and rethrow it at the end. some exceptions may be handled immediately and then swallowed,
+        // as such we just need to skip those dirty tasks in the checkpoint
+        final Set<TaskId> dirtyTaskIds = new HashSet<>();
         try {
             commitOffsetsOrTransaction(consumedOffsetsPerTask);
+        } catch (final TaskCorruptedException e) {
+            log.warn("Some tasks were corrupted when trying to commit offsets, these will be cleaned and revived: {}",
+                     e.corruptedTasks());
+
+            // If we hit a TaskCorruptedException, just handle the cleanup for those corrupted tasks right here
+            dirtyTaskIds.addAll(e.corruptedTasks());
+            closeDirtyAndRevive(tasks.tasks(dirtyTaskIds), true);
+        } catch (final TimeoutException e) {
+            log.warn("Timed out while trying to commit all tasks during revocation, these will be cleaned and revived");
+
+            // If we hit a TimeoutException we can just close dirty and revive without wiping the state
+            closeDirtyAndRevive(tasks.activeTasks(), false);
+            dirtyTaskIds.addAll(tasks.activeTaskIds());
+
+            try {
+                tasks.activeTasks().forEach(t -> t.maybeInitTaskTimeoutOrThrow(time.milliseconds(), e));
```

Comment:
As we handle revocation and thus don't own this task any longer, it might be ok to skip this step? What do we gain if we kill the thread?